src/Microsoft.DotNet.DeadRegionAnalysis/AnalysisEngine.Options.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.MSBuild;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.DeadRegionAnalysis
+{
+    public partial class AnalysisEngine
+    {
+        public class Options
+        {
+            private IEnumerable<IReadOnlyDictionary<string, Tristate>> _symbolConfigurations;
+            private Tristate _undefinedSymbolValue;
+
+            public IEnumerable<Document> Documents { get; private set; }
+
+            internal Options(
+                IEnumerable<Project> projects = null,
+                IEnumerable<string> projectPaths = null,
+                IEnumerable<string> sourcePaths = null,
+                IEnumerable<IEnumerable<string>> symbolConfigurations = null,
+                IEnumerable<string> alwaysIgnoredSymbols = null,
+                IEnumerable<string> alwaysDefinedSymbols = null,
+                IEnumerable<string> alwaysDisabledSymbols = null,
+                Tristate undefinedSymbolValue = default(Tristate))
+            {
+                if (projectPaths != null)
+                {
+                    projects = Task.WhenAll(from path in projectPaths select MSBuildWorkspace.Create().OpenProjectAsync(path, CancellationToken.None)).Result;
+                }
+                if (projects != null)
+                {
+                    Documents = GetSharedDocuments(projects);
+                }
+
+                if (sourcePaths != null)
+                {
+                    var projectId = ProjectId.CreateNewId("AnalysisProject");
+                    var solution = new AdhocWorkspace()
+                        .CurrentSolution
+                        .AddProject(projectId, "AnalysisProject", "AnalysisProject", LanguageNames.CSharp);
+
+                    foreach (var path in sourcePaths)
+                    {
+                        var documentId = DocumentId.CreateNewId(projectId);
+                        solution = solution.AddDocument(
+                            documentId,
+                            Path.GetFileName(path),
+                            new FileTextLoader(path, Encoding.UTF8));
```

Comment:
Using an explicit encoding here will break files with a non-UTF8 encoding.  We should just load with encoding detection enabled.